### PR TITLE
Use of highmem nodes for new clusters node pools and require a few tf variables to be set

### PR DESCRIPTION
**Pull request by consideRatio**
I've deliberated quite a bit on node pool's instance types, and I think these are good defaults for node pools in new k8s clusters to provide.

I've updated documentation a bit about the choices made for the core node pool specifically. The documentation is certainly not exhausting the discussion or is perfectly written, but I hope it can be seen as an incremental improvement.

### Summary of changes to node pools

GKE:
- `n2` is used systematically over `n1`. They provide a bit better performance per CPU and a 1:8 ratio of CPU:Memory instead of 1:6.5
- `n2-highmem-2` as core node pool for basehubs
- `n2-highmem-4` as core node pool for daskhubs
- `n2-highmem-4`, `-16`, and `-64` for user node pool for basehub and daskhub

EKS:
- `r5` is used systematically over `m5`. They provide a bit better performance per CPU and a 1:8 ratio of CPU:Memory instead of 1:4.
- `r5.xlarge` as core node pool for basehubs and daskhubs (equivalent to `n2-highmem-4`), where the smaller option `r5.large` allows for too few pods per node
- `r5.xlarge`, `r5.4xlarge`, and `r5.16xlarge` for user node pool for basehub and daskhub (equivalent to `n2-highmem-4`, `-16`, and `-64`)

### Related

- #2121
- #2005

### CODE REVIEW

File: docs/topic/infrastructure/cluster-design.md [hunk 1]
```diff
@@ -1,43 +1,93 @@
...
+        [250, 4],
+        [500, 5],
+        [1000, 6],
+        [1500, 7],
+        [2000, 8]
+      ]
+    }
+```
+
+On GKE clusters with network policy enforcement, we look to edit the
```
Comment (pnasrat) [inline docs/topic/infrastructure/cluster-design.md:76]: Can you clarify this - is this something that just happens through terraform, deployer, does it need manual intervention (if so document steps),

**Review by pnasrat — CHANGES_REQUESTED** [review]
Minor doc question - however my main concern is around deploying this with minimal disruption

What's the deployment/roll out impact of this - will it cause all existing core nodes to need to be upgraded (ie does terraform plan run clean for existing clusters or want to upgrade existing pools?  How do user nodes get upgraded to the right machine type with this - will it cause rescheduling?

If so is there a plan to incrementally roll out?

**Comment by consideRatio** [comment]
Thanks for reviewing @pnasrat!!

> What's the deployment/roll out impact of this - will it cause all existing core nodes to need to be upgraded (ie does terraform plan run clean for existing clusters or want to upgrade existing pools?

Super relevant question, I should have made it clear in the PR that this is not meant to influence `terraform plan` for any clusters existing node pools. If it does, its a non-intentend mistake to catch. In practice this just updates template files for new clusters, and the default value of `core_node_machine_type` - so, is this default used?

Apparently no, the defautl value for the `core_node_machine_type` variable isn't used. Its good I think, but it would been even better if it wasn't provided at all and instead was something required to be explicitly set by individual projects. I'll look into updating that for unused default values to reduce the complexity.

File: docs/topic/infrastructure/cluster-design.md (hunk 1 above)
Comment (consideRatio) [reply docs/topic/infrastructure/cluster-design.md:76]: I opened https://github.com/2i2c-org/infrastructure/issues/2490 to offload further clarifications into another PR, right now we don't have a procedure or agreement on what to do.

An explicit action point in the linked issue is to update this text once its done!

**Comment by consideRatio** [comment]
It seems by removing the default value specification, it becomes required! I'll do so.

> If present, the variable is considered to be optional and the default value will be used if no value is set when calling the module or running Terraform.

**Review by pnasrat — APPROVED** [review]
lgtm

**Comment by consideRatio** [comment]
Thank you @pnasrat!!

**Comment by damianavila** [comment]
> Super relevant question, I should have made it clear in the PR that this is not meant to influence terraform plan for any clusters existing node pools. If it does, its a non-intentend mistake to catch.

Is there any next intention/step to move the existing clusters to this new default, @consideRatio? If the answer is yes (I presume that is the case given #2121 context), is that captured in a new issue?

**Comment by consideRatio** [comment]
@damianavila I don't have a clear next step that I think we should actively work by itself, but I think its relevant to opportunistically consider transitioning to highmem nodes when perform disruptive maintenance.

I opened https://github.com/2i2c-org/infrastructure/issues/2511 about this!

**Pull request merged**